emails/models.py

```diff
@@ -211,11 +210,29 @@ def make_domain_address(user, address=None):
             # FIXME: Should we restrict users to create alias with bad words?
             raise CannotMakeAddressException
         address_hash = sha256(
-            '{}@{}'.format(domain_address.address, user_subdomain).encode('utf-8')
+            '{}@{}'.format(address, user_subdomain).encode('utf-8')
         ).hexdigest()
         address_already_deleted = DeletedAddress.objects.filter(
             address_hash=address_hash
         ).count()
         if address_already_deleted > 0:
             raise CannotMakeAddressException
+        domain_address = DomainAddress.objects.create(user=user, address=address)
         return domain_address
+
+    def delete(self, *args, **kwargs):
+        # TODO: create hard bounce receipt rule in AWS for the address
+        profile = Profile.objects.get(user=self.user)
+        deleted_address = DeletedAddress.objects.create(
+            address_hash=sha256(
+                f'{self.address}@{profile.subdomain}'.encode('utf-8')
+            ).hexdigest(),
```

Comment:
Let's make this into a `self.hash` property and re-use it here and in the `address_already_deleted` check in `make_domain_address` ?

Reply:
Couldn't use `self.hash` but instead used `address_hash` function to centralize the hash logic to a single place. See 7779b6d2686b4d707cf81f2b32eb92ea378ebd29